File: Others/insertLinkedLIst.java
package Others;

class DNode {

    int val;
    DNode next;
    DNode prev;

    DNode(int val) {
        this.val = val;
        this.next = null;
        this.prev = null;
    }
}

class DoublyLinkedList {

    DNode head;
    DNode tail;  // Track tail for efficiency
    int size;

    DoublyLinkedList() {
        head = null;
        tail = null;
        size = 0;
    }

    // Insert val at index (0 = head, -1 = end)
    void insert(int index, int val) {

    }

    // Get val at index
    int get(int index) {
        if (index < 0 || index >= size || head == null) {
            throw new IndexOutOfBoundsException();
        } else {
            // Traverse to node before insertion point
            DNode c1 = head;
            int count = 0;
            while (count < index) {
                c1 = c1.next;
                count++;
            }
            return c1.val;
        }

    }

    // Add val to end
    void add(int val) {
        DNode newNode = new DNode(val);
        if (size == 0) {
            head = newNode;
            tail = newNode;
        } else {
            newNode.prev = tail;
            tail.next = newNode;
            tail = newNode;
        }
        size++;

    }

    // Pop at index (default -1), return val
    int pop(int index) {
        if (size == 0) {
            return -1;
        } else {
            DNode c1 = head;
            int count = 0;
            while (count < index) {
                c1 = c1.next;
                count++;
            }
            DNode toremove = c1.next;
            c1.next = toremove.next;
            toremove.next.prev = c1;
            toremove.prev = null;
            toremove.next = null;

          

        }
        size--;
        return -1;
    }

    // Set val at index
    boolean set(int index, int val) {
        // Your code here
        return false;
    }

    // Helper: Print forward
    void print() {
        if (head == null) {
            System.out.println("Empty");
            return;
        }
        DNode curr = head;
        System.out.print(curr.val);
        curr = curr.next;
        while (curr != null) {
            System.out.print(" -> " + curr.val);
            curr = curr.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        // Same test as SLL
        DoublyLinkedList dll = new DoublyLinkedList();
        dll.add(1);
        dll.add(2);
        dll.insert(1, 10);
        dll.print();  // Expected: 1 -> 10 -> 2
    }
}

File: Others/insertLinkedList.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

class SinglyLinkedList:
    def __init__(self):
        self.head = None
        self.size = 0  # Track length

    # Insert val at index (0 = head, -1 = end)
    def insert(self, index, val):
        # Your code here (from previous)
        pass

    # Get val at index (return None if out of bounds)
    def get(self, index):
        if (index < 0 | index > list.size):
        
        pass

    # Add val to end
    def add(self, val):
        # Your code here
        pass

    # Pop (remove) at index, return val (default index=-1 for end)
    def pop(self, index=-1):
        # Your code here
        pass

    # Set val at index (return True if success, False if out of bounds)
    def set(self, index, val):
        # Your code here
        pass

    # NEW: Insert entire other_list after index (0=after head, -1=after end)
    def insertList(self, index, other_list):
        # Your code here: Splice other_list after node at index
        # Update self.size += other_list.size
        # If other_list is empty (head=None), do nothing
        pass

    # Helper: Print list
    def __str__(self):
        if not self.head:
            return "Empty"
        vals = []
        curr = self.head
        while curr:
            vals.append(str(curr.val))
            curr = curr.next
        return " -> ".join(vals)

# Test previous methods + new one
sll = SinglyLinkedList()
sll.add(1)
sll.add(3)
sll.add(5)  # Now: 1 -> 3 -> 5

sll2 = SinglyLinkedList()
sll2.add(2)
sll2.add(4)
sll2.add(6)  # Now: 2 -> 4 -> 6

sll.insertList(1, sll2)  # Insert sll2 after index 1 (after 3): 1 -> 3 -> 2 -> 4 -> 6 -> 5
print(sll)  # Expected: 1 -> 3 -> 2 -> 4 -> 6 -> 5

# Test empty other_list
sll3 = SinglyLinkedList()  # Empty
sll.insertList(0, sll3)  # Should do nothing
print(sll)  # Still: 1 -> 3 -> 2 -> 4 -> 6 -> 5